### DISCUSSION TITLE
What are some good databases to use with SvelteKit?

### DISCUSSION THREAD

**Discussion by FunctionalMetatable** [sveltejs/kit#3103] [Q&A]
I've been having problems with mongodb (monk) and I was thinking on moving to a different database (e.g Postgres).

**Comment by bill-kerr** [comment]
Hard to go wrong with Postgres + Prisma (https://www.prisma.io/).

Prisma can be used with quite a few databases though.

**Reply by cheluskin** [reply]
Prisma is not intended for production. Try npm run build and get errors with it.

**Reply by Glench** [reply]
Well, Prisma is absolutely intended for production use. Whether it works with SvelteKit currently is another matter.

**Reply by garytube** [reply]
Prisma does not support ESM, thats why you build fail
you need to treat the import differently when you do your production build

that's how i do it
`
// /lib/db.ts
import { dev } from '$app/env';
import * as Prisma from '@prisma/client';

// FOR PROD
import { default as PrismaProd } from '@prisma/client';

let { PrismaClient } = Prisma;
if (!dev) PrismaClient = PrismaProd.PrismaClient;

export default new PrismaClient({
	// log: ['query', 'warn', 'error']
});
`

**Comment by benmccann** [comment]
The choice of database should depend more on your use case than what frontend framework you are using

**Comment by leo-at-itartpro** [comment]
I like to use sqlite for all small projects, mariadb for cheap shared hosting projects (because usually shared hosting comes with it installed), and postgresql for everything with a lot of data.

**Reply by ghost** [reply]
How do you host your SvelteKit apps with SQLite for a database?

**Reply by Glench** [reply]
I host my app on DigitalOcean with a sqlite database, [LiteStream](https://litestream.io/) for streaming backups, SvelteKit, and [Sequelize](https://sequelize.org/). It works great — super performant so far, saved me money on database hosting, and has basically no operational setup/maintenance/costs. Though I'm not totally enamored with Sequelize.

**Reply by ghost** [reply]
How much roughly are you paying per month?

**Reply by Glench** [reply]
$6

**Comment by Glench** [comment]
It always depends on your specific needs, but for ease of deployment and performance [Consider SQLite](https://blog.wesleyac.com/posts/consider-sqlite). Postgres is probably one of the most popular databases, though.

**Comment by codepainting** [comment]
There is no good answer to that question other than “it depends”. There are many databases with different concepts behind them, solving different problems.

That said, I could imagine a concept in the spirit of SvelteKit adapters applied also to DBaaS providers, fine-tuned for the respective solutions (e.g. `adapter-db-supabase`, `adapter-db-atlas` etc.).

On a side-note: I trust our developers team, and they are not so much into the ORM side of things, simply because there are no great-performing one-fits-all-solutions. An adapter on the other hand could expose all advantages of the selected database system without that much overhead.

For the community + collaboration side of things, this could pose an interesting concept to get those DBaaS providers onboard and engage them to deliver the best DX possible for their respective platforms and SvelteKit at the same time.

**Reply by jesperordrup** [reply]
Postgres + knex seems to work really well

**Comment by Krowemoh** [comment]
Using better-sqlite3 and writing SQL directly has been great so far.
